File: Classes/GPUImageYoFilter.m
#import "GPUImageYoFilter.h"

@implementation GPUImageYoFilter

- (id)init;
{
    if (!(self = [super init]))
    {
        return nil;
    }
    
    // Blue Valentine
    GPUImageSaturationFilter *saturFilter = [[GPUImageSaturationFilter alloc] init];
    [saturFilter setSaturation:0.5];
    [self addFilter:saturFilter];
    
    GPUImageMonochromeFilter *monoFilter = [[GPUImageMonochromeFilter alloc] init];
    [monoFilter setColor:(GPUVector4){0.0f, 3.0f, 2.0f, 1.0f}];
    [monoFilter setIntensity:0.1];
    [self addFilter:monoFilter];
    
    GPUImagePixellateFilter *vigFilter = [[GPUImagePixellateFilter alloc] init];
    vigFilter.fractionalWidthOfAPixel = 0.8;
    [self addFilter:vigFilter];
    
    GPUImageExposureFilter *expoFilter = [[GPUImageExposureFilter alloc] init];
    [expoFilter setExposure:0.3];
    [self addFilter:expoFilter];
    
    [saturFilter addTarget:monoFilter];
    [monoFilter addTarget:vigFilter];
    [vigFilter addTarget:expoFilter];
    
    [self setInitialFilters:[NSArray arrayWithObject:saturFilter]];
    [self setTerminalFilter:expoFilter];
    
    return self;
}

@end

File: Classes/GPUImageYoFilter.swift
//
//  GPUImageYoFilter.swift
//  moment
//
//  Created by Nuno Gonçalves on 23/04/16.
//  Copyright © 2016 Motionblink. All rights reserved.
//

import UIKit
import GPUImage

class GPUImageYoFilter: GPUImageFilterGroup
{
    override init()
    {
        super.init()
        
        let saturFilter = GPUImageSaturationFilter()
        saturFilter.saturation = 0.5
        
        let monoFilter = GPUImageMonochromeFilter()
        monoFilter.color = GPUVector4(one: 0, two: 3, three: 2, four: 1)
        monoFilter.intensity = 0.1
        
        let vigFilter = GPUImageVignetteFilter()
        vigFilter.vignetteEnd = 0.8
        
        let expoFilter = GPUImageExposureFilter()
        expoFilter.exposure = 0.3
        
        saturFilter.addTarget(monoFilter)
        monoFilter.addTarget(vigFilter)
        vigFilter.addTarget(expoFilter)
        
        self.initialFilters = [saturFilter]
        self.terminalFilter = expoFilter
    }
}
